torch/fx/symbolic_trace.py

```diff
@@ -307,13 +307,136 @@ def forward(*args, **kwargs):
             # Seems to be a mypy limitation: https://github.com/python/mypy/issues/2427
             torch.nn.Module.__getattr__ = module_getattr_wrapper  # type: ignore
             torch.nn.Module.__call__ = module_call_wrapper
+
+            orig_fns = _patch_wrapped_functions()
+
             self.create_node('output', 'output', (self.create_arg(fn(*args)),), {},
                              type_expr=fn.__annotations__.get('return', None))
         finally:
+            _unpatch_wrapped_functions(orig_fns)
             torch.nn.Module.__call__ = orig_call
             torch.nn.Module.__getattr__ = orig_getattr  # type: ignore
         return self.graph
 
+# List of pairs of (global dict, function name) functions
+# to patch for the purposes of the wrap() API.
+_wrapped_fns_to_patch : List[Tuple[dict, str]] = []
+
+def _call_wrapped_func(orig_fn, args, kwargs):
+    """
+    Given an ``orig_function`` to invoke, search the args and kwargs for
+    a Proxy object. If there is one, emit a ``call_function`` node to
+    preserve the call to this leaf function directly. Otherwise, just
+    return the results of this function call, as this function is not
+    being traced.
+    """
+    proxy = None
+
+    def find_proxy(x):
+        nonlocal proxy
+        if isinstance(x, Proxy):
+            proxy = x
+
+    map_aggregate(args, find_proxy)
+    map_aggregate(kwargs, find_proxy)
+
+    if proxy is not None:
+        return proxy.tracer.create_proxy('call_function', orig_fn, args, kwargs)
+    else:
+        return orig_fn(*args, **kwargs)
+
+class PatchedFn(NamedTuple):
+    frame_dict : Dict[str, Any]
+    fn_name : str
+    orig_fn : Any
+
+def _patch_wrapped_functions() -> List[PatchedFn]:
+    """
+    Go through ``_wrapped_fn_patch_table`` and, for each frame object, wrap
+    the listed global functions in the `call_wrapped_func` wrapper. Returns
+    a list of PatchedFn, which is a record specifiying a single function
+    entry that was patched and contains the original function for unpatching
+    """
+    orig_fns : List[PatchedFn] = []
+    # Set to deduplicate entries. Wrapping a function multiple times would
+    # be an error, since it would cause a `call_function` node for the
+    # wrapper to be emitted rather than the actual underlying function
+    #
+    # Use id(frame_dict) as a hashable identity here since none of the
+    # frame dicts should be destroyed during symtracing
+    processed_entries : Set[Tuple[int, str]] = set()
+
+    for frame_dict, name in _wrapped_fns_to_patch:
+        if (id(frame_dict), name) in processed_entries:
+            continue
+        orig_fn = frame_dict[name]
+        orig_fns.append(PatchedFn(frame_dict, name, orig_fn))
+
+        def scope(orig_fn):
```

Comment:
If it was `_create_wrapped_func` rather than `_call_wrapped_func` this scoping hack would not be needed.